src/pages/workspaces/migration/migration-utils.ts

```diff
@@ -88,3 +120,54 @@ export const parseServerResponse = (
   }, sortedNamespaceKeys);
   return billingProjectWorkspaces;
 };
+
+export interface BillingProjectMigrationStats {
+  workspaceCount: number;
+  succeeded: number;
+  unscheduled: number;
+  errored: number;
+  inProgress: number;
+}
+
+export const getBillingProjectMigrationStats = (
+  billingProjectInfo: BillingProjectMigrationInfo
+): BillingProjectMigrationStats => {
+  const migrationStats: BillingProjectMigrationStats = {
+    workspaceCount: billingProjectInfo.workspaces.length,
+    unscheduled: 0,
+    errored: 0,
+    succeeded: 0,
+    inProgress: 0,
+  };
+
+  billingProjectInfo.workspaces.forEach((workspaceMigrationInfo) => {
+    Utils.cond(
+      [
+        workspaceMigrationInfo.migrationStep === 'Unscheduled',
+        () => {
+          migrationStats.unscheduled += 1;
+          return undefined;
```

Comment:
I'm not sure if there is a different way of calling `cond` (or some other thing) so that I don't need to return `undefined`.

Reply:
Huh, I thought I got TypeScript errors when I didn't have the explicit returns, but now it seems fine. I will remove them.